mkosi

```diff
@@ -1814,6 +1835,7 @@ def parse_args():
     group.add_argument('-C', "--directory", help='Change to specified directory before doing anything', metavar='PATH')
     group.add_argument("--default", dest='default_path', help='Read configuration data from file', metavar='PATH')
     group.add_argument("--kernel-commandline", help='Set the kernel command line (only bootable images)')
+    group.add_argument("--hostname", help="Set hostname")
```

Comment:
I think we should remove /etc/hostname if it is a regular file, the way the patch currently does it... That way we can take benefit of nspawns and systemd's hostname logic in a much nicer way: nspawn will initialize the container's hostname to the container's image/directory name (with suffixes removed), and if /etc/hostname isn#t there, then systemd will leave it that way, which is a good thing. If we'd however leave /etc/hostname in place, in particular with a default value of "localhost" or so, then this nspawn feature will never have any effect...

Also note that when systemd boots up and sees the hostname uninitialized, it will make sure it is initialized, and for that has a fallback built-in of "localhost" (but this can be changed by distros if they like at compile time, and fedora for example does, and changes this to "fedora"). Hence, leaving in /etc/hostname that only carries a default value is not only redudant in all cases but also hurtful in others...

Hence I think the way the patch does it right now is great.

(But yeah, you do have a point, and I actually would agree with you, if distros would get things right, and not drop in /etc/hostname by default in the first place. If they wouldn't there would of course be no point in deleting it...)